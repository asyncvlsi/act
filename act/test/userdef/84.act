deftype MyInnerType(int<16> a, b) {
    methods {
        function serialize() : int<32> {
            chp {
                self := a << 16 | b
            }
        }
    }
}

deftype MyOuterType(MyInnerType inner) {
    methods {
        function serialize() : int<32> {
            chp {
                self := inner.serialize()
            }
        }
    }
}

defproc test() {
    MyInnerType ti;
    MyOuterType to;
    chp {
        ti := MyInnerType(0x23 << 16 | 0x18);
        to := MyOuterType(ti);
        log("%x", ti.serialize());
        log("%x", to.serialize())
    }
}
