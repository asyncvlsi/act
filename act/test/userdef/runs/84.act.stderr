ERROR: File `84.act', line 26, col 27
	Typechecking failed on expression!
	Built-in macro `MyOuterType': argument has an incompatible type
